numpy/_core/tests/test_numeric.py

```diff
@@ -1956,6 +1956,29 @@ def __bool__(self):
         a = np.array([[ThrowsAfter(15)]] * 10)
         assert_raises(ValueError, np.nonzero, a)
 
+    def test_nonzero_byteorder(self):
+        values = [0., -0., 1, float('nan'), 0, 1,
+                  np.float16(0), np.float16(12.3)]
+        expected = [0, 0, 1, 1, 0, 1, 0, 1]
+
+        for value, expected in zip(values, expected):
+            A = np.array([value])
+            A_byteswapped = (A.view(A.dtype.newbyteorder()).byteswap()).copy()
+
+            assert np.count_nonzero(A) == expected
+            assert np.count_nonzero(A_byteswapped) == expected
+
+    def test_count_nonzero_non_aligned_array(self):
+        sz = 64
+        b = np.zeros(64 + 1).view(np.int8)[1:-(np.intp(0).itemsize - 1)]
```

Comment:
`intp` itemsize != float64 itemsize.  Just hardcode the 8/7 bytes, seems easier to read.  Can also create the `int8/uint8` array initially directly.